### PR TITLE
Implement Hash for Guid

### PR DESCRIPTION
**Pull request by jrmuizel**

### CODE REVIEW

File: src/runtime/guid.rs [hunk 1]
```diff
@@ -131,6 +132,15 @@ impl From<&str> for Guid {
     }
 }
 
+impl Hash for Guid {
```
Comment (kennykerr) [inline src/runtime/guid.rs:135]: Any reason not to simply derive this?

**Comment by kennykerr** [comment]
Thanks for the contribution @jrmuizel - generally it helps to open an issue or at least include a PR comment describing the motivation for the change. 

This particular change seems mostly self-explanatory but there's still the issue of consistency since we don't generate `Hash` for any other Windows structs at present. I think `Hash` could be useful to apply more broadly but it may incur an unexpected compilation cost when applied to all Windows structs. I just haven't done the homework to confirm yet. 

Also #589 for reference.

File: src/runtime/guid.rs (hunk 1 above)
Comment (jrmuizel) [reply src/runtime/guid.rs]: Nope, I forgot you could just do that. I've updated the PR.

**Comment by jrmuizel** [comment]
This PR was motivated by https://github.com/n4r1b/ferrisetw/blob/b631bcdd01bc5da250980092d1be192b826e4c02/src/schema.rs#L34

As for the consistency issue, GUIDs are closer to a primitive type on Windows vs other structs. Examples of this would be the `uuid` attribute in C++ and the GUID input type here: https://docs.microsoft.com/en-us/windows/win32/wes/eventmanifestschema-inputtype-complextype

**Review by kennykerr — APPROVED** [review]
Thanks!

**Pull request merged**